src/build_system.ml

```diff
@@ -222,19 +236,13 @@ module Alias0 = struct
     if not (Path.is_in_build_dir path) then
```

Comment:
I believe this test is no longer needed

Reply:
Yeah, but then we're changing the user error message in `die` into something more opaque. I think it's still possible for the user to provide an invalid path here (though I'm not sure).

Reply:
Maybe when writing `(alias /foo/bar)`? `Alias.of_path` seems to be only used for that. I suggest to change `of_path`  to `of_path_written_by_user : loc:Loc.t -> ...` and make it raise a properly located error.

Reply:
This is not directly related to this PR, but we might as well fix it once and for all

Reply:
Yeah, that's a good idea - though i think this is not a trivial change. The only way I can see how to do this is to change Dep_conf.t to preserve the locs. If it's that big, I might as well make it its own PR.

Reply:
The argument of `Dep_conf.Alias` is a `String_with_vars.t` which already contain a location

Reply:
Yeah, that wasn't too hard. Added a test for this error message too.

Reply:
Ok, looks good. One last thing: the checks done by `M.make` are  redundant here since we already check that the path is in the build directory and `Path.basename x` is guaranted not to contain `/`. Can you move `of_user_writen_path` in `M` and create the `t` value directly?

Reply:
Done.